File: TEST data for large repo/Repo Copy-3/Blueprism/BluePrism.Server.Domain.Models/WorkQueueItemParameters.vb
Imports System.Runtime.Serialization
Imports BluePrism.Server.Domain.Models.Pagination
Imports Func
Imports BluePrism.Server.Domain.Models.DataFilters

<Serializable()>
<DataContract([Namespace]:="bp")>
Public Class WorkQueueItemParameters
    <DataMember(Name:="s", EmitDefaultValue:=False)>
    Public Property SortBy As WorkQueueItemSortByProperty
    <IgnoreDataMember>
    Public Property KeyValue As DataFilter(Of String)
    <IgnoreDataMember>
    Public Property Status As DataFilter(Of String)
    <IgnoreDataMember>
    Public Property ExceptionReason As DataFilter(Of String)
    <IgnoreDataMember>
    Public Property WorkTime As DataFilter(Of Integer)
    <IgnoreDataMember>
    Public Property Attempt As DataFilter(Of Integer)
    <IgnoreDataMember>
    Public Property Priority As DataFilter(Of Integer)
    <IgnoreDataMember>
    Public Property LoadedDate As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property LastUpdated As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property DeferredDate As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property LockedDate As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property CompletedDate As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property ExceptionedDate As DataFilter(Of DateTimeOffset)
    <IgnoreDataMember>
    Public Property ItemsPerPage As Integer
    <IgnoreDataMember>
    Public Property PagingToken As [Option](Of WorkQueueItemPagingToken)
End Class

File: TEST data for large repo/repo_Copy-2/Blueprism/BluePrism.Api/BluePrism.Api.Models/WorkQueueItemParameters.cs
namespace BluePrism.Api.Models
{
    using System;

    public class WorkQueueItemParameters : IPagingModel<long>
    {
        public WorkQueueItemSortBy? SortBy { get; set; }
        
        public StartsWithOrContainsStringFilterModel KeyValue { get; set; }

        public StartsWithOrContainsStringFilterModel Status { get; set; }

        public StartsWithOrContainsStringFilterModel ExceptionReason { get; set; }

        public RangeFilterModel<int?> TotalWorkTime { get; set; }

        public RangeFilterModel<int?> Attempt { get; set; }

        public RangeFilterModel<int?> Priority { get; set; }

        public RangeFilterModel<DateTimeOffset?> LoadedDate { get; set; }

        public RangeFilterModel<DateTimeOffset?> DeferredDate { get; set; }

        public RangeFilterModel<DateTimeOffset?> LockedDate { get; set; }

        public RangeFilterModel<DateTimeOffset?> CompletedDate { get; set; }

        public RangeFilterModel<DateTimeOffset?> ExceptionedDate { get; set; }

        public RangeFilterModel<DateTimeOffset?> LastUpdated { get; set; }

        public int? ItemsPerPage { get; set; }

        public PagingTokenModel<long> PagingToken { get; set; }
    }
}
